src/msw/listctrl.cpp

```diff
@@ -1494,12 +1499,31 @@ bool wxListCtrl::EnableCheckBoxes(bool enable)
 
 void wxListCtrl::CheckItem(long item, bool state)
 {
-    ListView_SetCheckState(GetHwnd(), (UINT)item, (BOOL)state);
+    if ( !HasCheckBoxes() )
+        return;
```

Comment:
Should this be a `wxCHECK_RET()` instead? I.e. isn't it a programmers error to call this if there are no checkboxes?

Reply:
Yes, I added it.